### PR TITLE
Update for Bullet Shower to 4.0-dev

### PR DESCRIPTION
**Pull request by voylin**
Fixed some issues of things which changed in Godot 4 and I also tried replicating this issue #599 but I guess that is not longer an issue in Godot 4.0? Tested and fixed everything on Godot 4.0 alpha-5.

### CODE REVIEW

File: 2d/bullet_shower/README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ in the documentation for more information.
 
 Language: GDScript
 
-Renderer: GLES 2
+Renderer: Vulkan
```
Comment (aaronfranke) [inline 2d/bullet_shower/README.md:12]: If we're updating this text, we should be specifying if it's Vulkan Clustered or Vulkan Mobile.

Additionally, we should be setting demos to Vulkan Mobile if it doesn't look any different compared to Vulkan Clustered. Especially for GUI-only demos, they will look the same in Vulkan Mobile. Go to Project -> Project Settings -> General tab -> Rendering section -> Vulkan -> Back End -> Select Forward Mobile.

**Review by aaronfranke — COMMENTED** [review]
Most of this looks good.

File: 2d/bullet_shower/bullets.gd [hunk 2]
```diff
@@ -4,13 +4,13 @@ extends Node2D
...
 
-const BULLET_COUNT = 500
-const SPEED_MIN = 20
-const SPEED_MAX = 80
+const BULLET_COUNT := 500
+const SPEED_MIN := 20
+const SPEED_MAX := 80
 
-const bullet_image = preload("res://bullet.png")
+const bullet_image := preload("res://bullet.png")
```
Comment (aaronfranke) [inline 2d/bullet_shower/bullets.gd:11]: Constants are implicitly statically typed, there's no purpose to the `:=`.

File: 2d/bullet_shower/shower.tscn [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
 "loop": true,
-"name": "default",
+"name": &"default",
 "speed": 5.0
 }]
 
-[sub_resource type="CircleShape2D" id=2]
-radius = 27.0
+[sub_resource type="CircleShape2D" id="2"]
+radius = 28.0179
```
Comment (aaronfranke) [inline 2d/bullet_shower/shower.tscn:17]: Why did this change?

Comment (voylin) [reply 2d/bullet_shower/shower.tscn]: I did some testing to find out why it wasn't working and forgot to change it back to 27. Fixed all the things you have mentioned, squashed the commits and pushed. Thanks again for teaching me how to squash commits! It's very helpful to stay organized. ^^

File: 2d/bullet_shower/bullets.gd (hunk 2 above)
Comment (aaronfranke) [reply 2d/bullet_shower/bullets.gd:11]: This line can still be simplified.

Comment (voylin) [reply 2d/bullet_shower/bullets.gd:11]: Sorry, forgot about that one. fixed, merged and pushed!

Comment (aaronfranke) [reply 2d/bullet_shower/bullets.gd:11]: It doesn't look like you did. Did you forget to stage your changes before clicking amend and push? Happens to me a lot.

Comment (voylin) [reply 2d/bullet_shower/bullets.gd]: hahah I think that must have happened, sorry about that.

**Review by aaronfranke — APPROVED** [review]
Thanks!

**Pull request merged**